jme3-core/src/main/java/com/jme3/renderer/pipeline/DefaultPipelineContext.java

```diff
@@ -0,0 +1,60 @@
+/*
+ * Copyright (c) 2024 jMonkeyEngine
+ * All rights reserved.
+ *
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are permitted provided that the following conditions are
+ * met:
+ *
+ * * Redistributions of source code must retain the above copyright
+ *   notice, this list of conditions and the following disclaimer.
+ *
+ * * Redistributions in binary form must reproduce the above copyright
+ *   notice, this list of conditions and the following disclaimer in the
+ *   documentation and/or other materials provided with the distribution.
+ *
+ * * Neither the name of 'jMonkeyEngine' nor the names of its contributors
+ *   may be used to endorse or promote products derived from this software
+ *   without specific prior written permission.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+ * "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED
+ * TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR
+ * PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT OWNER OR
+ * CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL,
+ * EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO,
+ * PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR
+ * PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF
+ * LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING
+ * NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS
+ * SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+package com.jme3.renderer.pipeline;
+
+import com.jme3.renderer.RenderManager;
+import com.jme3.renderer.ViewPort;
+import java.util.concurrent.atomic.AtomicBoolean;
+
+/**
+ * Default implementation of AbstractPipelineContext that
+ * does nothing extra.
+ * 
+ * @author codex
+ */
+public class DefaultPipelineContext implements PipelineContext {
+
+    // decided to use an atomic boolean, since it is less hassle
+    private final AtomicBoolean rendered = new AtomicBoolean(false);
+    
+    @Override
+    public boolean startViewPortRender(RenderManager rm, ViewPort vp) {
+        return rendered.getAndSet(true);
+    }
+    @Override
+    public void endViewPortRender(RenderManager rm, ViewPort vp) {}
```

Comment:
Please leave a blank line between methods to make the code more readable. In general it is a good practice to favor readability over compactness, as well as to respect the style of code adopted by jme. Please format the code for this class correctly.